x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -555,19 +560,19 @@ public void updateConnectorNameOrDescription(UpdateConnectorNameAction.Request r
     }
 
     /**
-     * Updates the {@link ConnectorFiltering} property of a {@link Connector}.
+     * Sets the {@link ConnectorFiltering} property of a {@link Connector}.
      *
-     * @param request   Request for updating connector filtering property.
-     * @param listener  Listener to respond to a successful response or an error.
+     * @param connectorId The ID of the {@link Connector} to update.
+     * @param filtering   The list of {@link ConnectorFiltering} .
+     * @param listener    Listener to respond to a successful response or an error.
      */
-    public void updateConnectorFiltering(UpdateConnectorFilteringAction.Request request, ActionListener<UpdateResponse> listener) {
+    public void updateConnectorFiltering(String connectorId, List<ConnectorFiltering> filtering, ActionListener<UpdateResponse> listener) {
```

Comment:
Would it be possible to update filtering through one method, regardless whether it's about updating `active` or `draft` filtering?

Having two methods `updateConnectorFiltering` and `updateConnectorFilteringDraft` without having `updateConnectorFilteringActive` _feels_ at a first sight like something, which can be simplified.

Reply:
`updateConnectorFiltering` has different behaviour (since it's the older method that overrides all filtering object). 

`updateConnectorFilteringDraft` should be preferred method and has a lot of logic and checks that are not present in `updateConnectorFiltering` - I prefer to keep them separate for easier testing. IMO `updateConnectorFiltering` should be deleted altogether in the future as we don't want people to update all filtering object at once.